### PR TITLE
Search: Fix only searching for folder id zero

### LINKED ISSUE #40273: Search: Using folder id 0 seems to return all folders incorrectly
**What happened**:
Given the https://github.com/grafana/grafana/pull/40192#issuecomment-938417252 implies searching for `folderIds=0` returns all the folders and dashboards.

**What you expected to happen**:
Only the general folder and its dashboards to be returned.

**Environment**:
- Grafana version: 8.2.0

### PR DESCRIPTION
**Pull request by marefr**
**What this PR does / why we need it**:
Fixes so that searching for folder id zero in folder/dashboard search returns dashboards located in the general folder and not including all folders as it did before.

**Which issue(s) this PR fixes**:
Fixes #40273 
Ref #40192

**Special notes for your reviewer**:
- Calling `http://localhost:3000/api/search?folderIds=0` should only return dashboards located in that folder.

# Release notice breaking change

Searching for only folder id zero in folder/dashboard search (`/api/search?folderIds=0`) now returns dashboards located in the general folder and not including all folders as it did before.

### CODE REVIEW

**Review by hugohaggmark — COMMENTED** [review]
LGTM 🎉  any tests that need to be written/updated?

**Review by hugohaggmark — APPROVED** [review]
LGTM 🎉  (but I'm no Go expert 🙈 )

**Review by kalleep — APPROVED** [review]
LGTM

File: pkg/services/sqlstore/searchstore/filters.go [hunk 1]
```diff
@@ -87,10 +87,16 @@ func (f TitleFilter) Where() (string, []interface{}) {
 }
 
 type FolderFilter struct {
-	IDs []int64
+	Dialect migrator.Dialect
+	IDs     []int64
 }
 
 func (f FolderFilter) Where() (string, []interface{}) {
+	if len(f.IDs) == 1 && f.IDs[0] == 0 {
```
Comment (DanCech) [inline pkg/services/sqlstore/searchstore/filters.go:95]: what happens if `f.IDs` is `[0,1]`? Is that a use-case we need to worry about?

Comment (marefr) [reply pkg/services/sqlstore/searchstore/filters.go:95]: It might be same problem, but don't think it's a use case to worry about for the moment.

**Comment by mure** [comment]
Hi, total newbie here to any Grafana development, but I think this change may have caused a problem in the UI. It seems by default the dashboard browsing view (/dashboards) makes this call:
`http://localhost:3000/api/search?query=&starred=false&skipRecent=true&skipStarred=true&folderIds=0&layout=folders&prevSort=null`

So now that folderIds=0 _only_ returns the general folder, you don't see any of your other folders 😬

**Comment by marefr** [comment]
@mure thanks. Have been looking into this and decided to revert these changes since it introduced the problem you referred to and other unexpected side effects.

**Pull request merged**